### DISCUSSION TITLE
Settings PropertyEditorUI for mapping fields.

### DISCUSSION THREAD

**Discussion by enkelmedia** [umbraco/Umbraco.Forms.Issues#1525] [Ideas]
I often tend to need a way to map known properties against fields in the form.

Some examples:

**Integrate with CRM X**
In this scenario we know that we want to collect Firstname, Lastname, Company and Email, on the workflow I would like to offer the editor to choose the mapping in a nice way.

**Integrate with Campaign tool Y**
In this scenario we know that we want to collect Name and Email, on the workflow I would like to offer the editor to choose the mapping in a nice way.

It would be really nice if there was a Property Editor UI that we could feed with these "known" properties and that would show them together with a form field dropdown so the editor can "map them".

Basically a table like so:

| Property | Field |
|-----|-----|
| Firstname | --Choose Field-- |
| Lastname | --Choose Field-- |
| Company | --Choose Field-- |
| Email | --Choose Field-- |

I did look at the docs https://docs.umbraco.com/umbraco-forms/developer/extending/adding-a-type/setting-types but as I understand it there is no such thing today?

The closest thing I could find was `forms-field-mapper-property-editor` but here the "left side" is not predefined. 

![image](https://github.com/user-attachments/assets/0b52270a-bc94-43b0-ab1d-9ac22c34b5b4)

I'm basically looking for something like this but where the left side i a predefined property with maybe alias (for code) and label (for UI).

**Comment by TvGessel** [comment]
Thank you for the suggestion. We have added this to our internal backlog for further discussion and prioritization to determine if and when we can include this in the roadmap.
